go/kbfs/libfs/util.go

```diff
@@ -48,3 +52,74 @@ func RecursiveDelete(
 
 	return fs.Remove(fi.Name())
 }
+
+var obsConflictRegexp = regexp.MustCompile(`-([[:digit:]]+)(\.|$)`)
+
+func stripObfuscatedConflictSuffix(s string) string {
+	replace := ""
+	if strings.Contains(s, ".") {
+		replace = "."
+	}
+	return obsConflictRegexp.ReplaceAllString(s, replace)
+}
+
+func deobfuscate(
+	ctx context.Context, fs *FS, p []string) (res []string, err error) {
+	if len(p) == 0 {
+		return nil, nil
+	}
+
+	fis, err := fs.ReadDir("")
+	if err != nil {
+		return nil, err
+	}
+
+	elem := stripObfuscatedConflictSuffix(p[0])
+	for _, fi := range fis {
+		name := fi.Name()
+		obsName := stripObfuscatedConflictSuffix(fs.PathForLogging(name))
+		if obsName == elem {
+			if len(p) == 1 || !fi.IsDir() {
```

Comment:
I think you don't want the `|| !fi.IsDir()` here. It means that if `a` and `b` hash to the same thing, a deobfuscate that should yield `b/c` would also yield `a`.

Reply:
@jakob223 	Sorry I don't follow.  The check right above this is `obsName == elem`, which means this code will only be run on things that already match.  This is just putting filenames directly into the results, instead of trying to chroot into them.

Reply:
if `a` and `b` both obfuscate to `cloth-matrix`, then this check will catch `a` even when the input is `cloth-matix/coin-upset` unless I'm misunderstanding something.

Reply:
Ohh I see what you're saying.  `a` is a file and `b` is a directory, they're both in the same subdirectory, and we're trying to deobfuscate something that's under `b/`.  Ok you're right, I can drop that clause. One sec.

Reply:
```$ keybase fs debug deobfuscate /keybase/private/jakob223,strib/just-treat/simple-onion                                                                                                                               
/keybase/private/jakob223,strib/file304/hello
/keybase/private/jakob223,strib/file_3017```

Reply:
Ok done, please check it out.

Reply:
And nice catch, by the way!